balancer/state.go

```diff
@@ -0,0 +1,165 @@
+package balancer
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/R-a-dio/valkyrie/config"
+	"github.com/R-a-dio/valkyrie/errors"
+)
+
+// Balancer represents the state of the load balancer.
+type Balancer struct {
+	config.Config
+	manager radio.ManagerService
+	storage radio.StorageService
+	serv    *http.Server
+
+	// The current stream to re-direct clients to.
+	current atomic.Value
```

Comment:
methods added are nice, but should be a different type such that no code actually has realistic access to the `atomic.Value`

Reply:
i addressed this by adding a package named `current` that wraps a private `atomic.Value` with Get and Set methods.